src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react'
+import { useParams, Link, useNavigate } from 'react-router-dom'
+import { getArticleById } from "../api";
+import { Share, MessageCircle, ArrowLeft, ArrowBigUp, ArrowBigDown } from 'lucide-react';
+import { formatDate } from '../../utils/formatting';
+
+const SingleArticle = () => {
+  const navigate = useNavigate();
+  const { article_id } = useParams()
+  const [article, setArticle] = useState([])
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getArticleById(article_id).then((articleData) => {
+      setArticle(articleData)
+      setIsLoading(false);
+    })
+    .catch((err) => {
+      console.error("Error fetching article:", err);
+      setError("Failed to load article");
+      setIsLoading(false);
+    });
+  }, [article_id])
+
+  if (isLoading) {
+    return (
+      <div className="loading-container">
+        <p className="loading-message">Loading...</p>
+      </div>
+    );
+  }
+  
+  if (error) return <p>{error}</p>;
+  if (!article) return <p>No article found</p>;
+
+  return (
+    <div className="single-article">
```

Comment:
Rather than navigating to comments here, it would be more conventional in React to have a piece of state called commentsVisible. We could then use this state to conditionally render the comments removing the need to navigate and creating a more conventional react experience